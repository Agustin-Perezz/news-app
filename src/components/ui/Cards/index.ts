export * from './ListCards';
export * from './customStyleCard';
