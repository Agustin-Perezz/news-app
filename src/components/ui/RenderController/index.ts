export * from './LoadingSkeleton';
export * from './RenderController';
