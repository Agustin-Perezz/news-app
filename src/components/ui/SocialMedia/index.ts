export * from './SocialMedia';
export * from './data-media';
