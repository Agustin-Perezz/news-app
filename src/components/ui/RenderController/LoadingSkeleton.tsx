import {
  Box,
  Grid,
  Skeleton,
  Card,
  CardContent,
  CardMedia,
  Fade,
} from '@mui/material';
import { CustomGrid } from '../Cards';

export const LoadingSkeleton: React.FC = () => {
  return (
    <Box
      sx={{
        display: 'flex',
        justifyContent: 'center',
        flexWrap: 'wrap',
      }}
    >
      <Grid sx={{ flexGrow: 1, margin: '25px 0px' }}>
        <Grid
          item
          xs={12}
          sx={{
            justifyContent: 'center',
            display: 'flex',
            position: 'relative',
          }}
        >
          <CustomGrid container spacing={3}>
            {Array(5)
              .fill(0)
              .map((__, index) => (
                // eslint-disable-next-line react/no-array-index-key
                <Grid key={index} item>
                  <Fade timeout={1500} in>
                    <Card sx={{ height: '448px', width: '278px' }}>
                      <CardMedia>
                        <Skeleton
                          variant="rectangular"
                          animation="wave"
                          width="100%"
                          height="170px"
                        />
                      </CardMedia>
                      <CardContent>
                        <Skeleton
                          variant="rounded"
                          animation="wave"
                          width="90%"
                          height="25px"
                          sx={{ marginTop: '10px' }}
                        />
                        <Skeleton
                          animation="wave"
                          width="100%"
                          height="18px"
                          sx={{ marginY: '5px', marginTop: '18px' }}
                        />
                        <Skeleton animation="wave" width="90%" height="18px" />
                        <Box
                          sx={{
                            display: 'flex',
                            justifyContent: 'flex-start',
                            gap: '5px',
                            marginTop: '15px',
                          }}
                        >
                          <Skeleton
                            animation="wave"
                            width="20%"
                            height="18px"
                          />
                          <Skeleton
                            animation="wave"
                            width="40%"
                            height="18px"
                          />
                        </Box>
                      </CardContent>
                    </Card>
                  </Fade>
                </Grid>
              ))}
          </CustomGrid>
        </Grid>
      </Grid>
    </Box>
  );
};
