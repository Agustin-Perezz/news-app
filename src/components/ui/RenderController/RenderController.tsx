import React from 'react';
import { NewsProps } from '../../../types/news-props';
import { ErrorBox } from '../Error';
import { LoadingSkeleton } from './LoadingSkeleton';

export type PropsRender = {
  children: React.ReactNode;
  isLoading: boolean;
  isError?: string;
  data?: NewsProps[];
};

export const RenderController: React.FC<PropsRender> = ({ children, data, isLoading, isError }) => {
  if (isLoading) return <LoadingSkeleton />;

  if (isError) return <ErrorBox errorMessage={isError} />;

  if (data?.length === 0) return <div>Ups, no results, try with others things.</div>;

  return <div>{children}</div>;
};
