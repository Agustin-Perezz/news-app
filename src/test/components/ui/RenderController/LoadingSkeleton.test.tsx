import { render, screen } from '@testing-library/react';
import { LoadingSkeleton } from '../../../../components/ui/RenderController';

describe('Test in <LoadingSkeleton />', () => {
  it('should match snapshot', () => {
    const { container } = render(<LoadingSkeleton />);
    expect(container).toMatchSnapshot();
  });
  it('should contain material ui classes for skeleton components ', () => {
    const { container } = render(<LoadingSkeleton />);
    const classSkeletons = container.getElementsByClassName('MuiSkeleton-root');
    expect(classSkeletons[0].className).toBe(
      'MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1kcnzpi-MuiSkeleton-root'
    );
    expect(classSkeletons.length).toBe(30);
  });
});
