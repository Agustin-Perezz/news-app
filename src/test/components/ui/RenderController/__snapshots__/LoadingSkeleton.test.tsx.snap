// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <LoadingSkeleton /> should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-1dlcyqs"
  >
    <div
      class="MuiGrid-root css-ip5omw-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1xkf9b7-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-bzgqu1-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1shyln9-MuiPaper-root-MuiCard-root"
              style="opacity: 1; webkit-transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
            >
              <div
                class="MuiCardMedia-root css-16cvb1l-MuiCardMedia-root"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1kcnzpi-MuiSkeleton-root"
                  style="width: 100%; height: 170px;"
                />
              </div>
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-wave css-13amv01-MuiSkeleton-root"
                  style="width: 90%; height: 25px;"
                />
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-4s3xnn-MuiSkeleton-root"
                  style="width: 100%; height: 18px;"
                />
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                  style="width: 90%; height: 18px;"
                />
                <div
                  class="MuiBox-root css-1h0nyql"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                    style="width: 20%; height: 18px;"
                  />
                  <span
                    class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                    style="width: 40%; height: 18px;"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1shyln9-MuiPaper-root-MuiCard-root"
              style="opacity: 1; webkit-transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
            >
              <div
                class="MuiCardMedia-root css-16cvb1l-MuiCardMedia-root"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1kcnzpi-MuiSkeleton-root"
                  style="width: 100%; height: 170px;"
                />
              </div>
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-wave css-13amv01-MuiSkeleton-root"
                  style="width: 90%; height: 25px;"
                />
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-4s3xnn-MuiSkeleton-root"
                  style="width: 100%; height: 18px;"
                />
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                  style="width: 90%; height: 18px;"
                />
                <div
                  class="MuiBox-root css-1h0nyql"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                    style="width: 20%; height: 18px;"
                  />
                  <span
                    class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                    style="width: 40%; height: 18px;"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1shyln9-MuiPaper-root-MuiCard-root"
              style="opacity: 1; webkit-transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
            >
              <div
                class="MuiCardMedia-root css-16cvb1l-MuiCardMedia-root"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1kcnzpi-MuiSkeleton-root"
                  style="width: 100%; height: 170px;"
                />
              </div>
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-wave css-13amv01-MuiSkeleton-root"
                  style="width: 90%; height: 25px;"
                />
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-4s3xnn-MuiSkeleton-root"
                  style="width: 100%; height: 18px;"
                />
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                  style="width: 90%; height: 18px;"
                />
                <div
                  class="MuiBox-root css-1h0nyql"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                    style="width: 20%; height: 18px;"
                  />
                  <span
                    class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                    style="width: 40%; height: 18px;"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1shyln9-MuiPaper-root-MuiCard-root"
              style="opacity: 1; webkit-transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
            >
              <div
                class="MuiCardMedia-root css-16cvb1l-MuiCardMedia-root"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1kcnzpi-MuiSkeleton-root"
                  style="width: 100%; height: 170px;"
                />
              </div>
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-wave css-13amv01-MuiSkeleton-root"
                  style="width: 90%; height: 25px;"
                />
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-4s3xnn-MuiSkeleton-root"
                  style="width: 100%; height: 18px;"
                />
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                  style="width: 90%; height: 18px;"
                />
                <div
                  class="MuiBox-root css-1h0nyql"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                    style="width: 20%; height: 18px;"
                  />
                  <span
                    class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                    style="width: 40%; height: 18px;"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1shyln9-MuiPaper-root-MuiCard-root"
              style="opacity: 1; webkit-transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
            >
              <div
                class="MuiCardMedia-root css-16cvb1l-MuiCardMedia-root"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1kcnzpi-MuiSkeleton-root"
                  style="width: 100%; height: 170px;"
                />
              </div>
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-wave css-13amv01-MuiSkeleton-root"
                  style="width: 90%; height: 25px;"
                />
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-4s3xnn-MuiSkeleton-root"
                  style="width: 100%; height: 18px;"
                />
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                  style="width: 90%; height: 18px;"
                />
                <div
                  class="MuiBox-root css-1h0nyql"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                    style="width: 20%; height: 18px;"
                  />
                  <span
                    class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                    style="width: 40%; height: 18px;"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
